tests/testthat/test-table-object.R

```diff
@@ -157,6 +157,16 @@ test_that("stobject equivalent rename", {
   expect_identical(x$output, y$output)
 })
 
+test_that("rename by named list", {
+  mt <- mtcars[1:3,]
+  y <- st_new(mt) %>%
+    st_rename(cylinder = cyl, .list = list(hp = "HorsePower")) %>%
```

Comment:
this is what I mean by "mixed both". I think the `.list` arg makes most sense as an alternative to the `...` so one or the other (and if you do use both,`.list` will be taken)

Also:

```
cylinder = cyl, .list = list(hp = "HorsePower")
```

This may lead to some headaches. In the `.list` you need to do old name = new name.. but in the `...` you are doing new name = old name. 

Changing it to the code I wrote above for `st_rename` would standardize these to old name = new name. And then this test would become:

```
test_that("rename by named list", {
  mt <- mtcars[1:3,]
  y <- st_new(mt) %>%
    st_rename(.list = list(hp = "HorsePower")) %>%
    inspect2()

  x <- st_new(mt) %>%
    st_rename("cyl" = "cylinder") %>%
    inspect2()
  
  # pass both ... and .list
  # should only change what's in .list
  z <- st_new(mt) %>%
    st_rename("cyl" = "cylinder", .list = list(hp = "HorsePower")) %>%
    inspect2()

  tst <- y$cols_new
  tst2 <- x$cols_new
  tst3 <- z$cols_new

  expect_identical(tst[4], "HorsePower")
  expect_identical(tst2[2], "cylinder")
  expect_identical(tst3[2], "cyl")
  expect_identical(tst3[4], "HorsePower")
})
```

Notice: if we go this route,  `...` has to be strings instead of symbols because we are getting rid of the `enquos`

Reply:
We are using `old-name = new-name` in `.list` because that is more natural for working with the yspec object.